src/hooks/useWeatherData.ts

```diff
@@ -0,0 +1,25 @@
+import { useQuery } from "@tanstack/react-query";
+import { getWeatherData } from "../api/weather.api";
+import useWeatherFormContext from "./useWeatherFromContext";
+
+export const useWeatherData = () => {
+  const { inputMethod, cityName, coordinates } = useWeatherFormContext();
+
+  const fetchWeather = async () => {
+    if (inputMethod === "cityName") {
+      return getWeatherData(cityName);
+    } else if (inputMethod === "coordinates") {
+      return getWeatherData(coordinates);
+    } else if (inputMethod === "currentLocation") {
+      return getWeatherData(coordinates);
+    } else {
+      throw new Error("Invalid input method");
+    }
+  };
+
+  return useQuery({
+    queryKey: ["weatherData", inputMethod, cityName, coordinates],
```

Comment:
You need to change this query key to a key that will actually remember your previous requests because for instance you can request for the city Paris and then you request for different coordinate and then you request Paris again, with this method you wont really bring the request from the caching